web/packages/teleport/src/Player/SshPlayer.tsx

```diff
@@ -87,35 +99,45 @@ const StyledPlayer = styled.div`
   justify-content: space-between;
 `;
 
-function useSshPlayer(clusterId: string, sid: string) {
-  const tty = React.useMemo(() => {
-    const prefixUrl = cfg.getSshPlaybackPrefixUrl({ clusterId, sid });
-    return new TtyPlayer(new EventProvider({ url: prefixUrl }));
-  }, [sid, clusterId]);
+function useStreamingSshPlayer(clusterId: string, sid: string) {
+  const [playerStatus, setPlayerStatus] = React.useState(StatusEnum.LOADING);
+  const [statusText, setStatusText] = React.useState('');
+  const [time, setTime] = React.useState(0);
 
-  // to trigger re-render when tty state changes
-  const [, rerender] = React.useState(tty.status);
+  const tty = React.useMemo(() => {
+    const url = cfg.api.ttyPlaybackWsAddr
+      .replace(':fqdn', getHostName())
+      .replace(':clusterId', clusterId)
+      .replace(':sid', sid)
+      .replace(':token', getAccessToken());
+    return new TtyPlayer({ url, setPlayerStatus, setStatusText, setTime });
+  }, [clusterId, sid, setPlayerStatus, setStatusText, setTime]);
 
   React.useEffect(() => {
-    function onChange() {
-      // trigger rerender when status changes
-      rerender(tty.status);
-    }
+    tty.connect(0, 0);
```

Comment:
I'm getting a type error here saying this expects no arguments. perhaps a left over?

Reply:
The base class expects `connect(width, height)`. I tried to leave them in the definition as `connect(_w, _h)` but then linter complained that those variables were unused. What's proper protocol here?

Reply:
We aren't calling the base `connect` in this method so we don't need to pass those params in. It overrides just fine with your overridden parameterless method declaration because the callsite is properly typed as `TtyPlayer` and knows its a parameterless `connect`. If i remove the `0,0` from this call of `tty.connect` all lint errors go away, is this different on your end?